qutebrowser/mainwindow/prompt.py

```diff
@@ -642,8 +643,16 @@ def _init_fileview(self):
         self._file_model.directoryLoaded.connect(
             lambda: self._file_model.sort(0))
 
+    def _show_error(self, msg):
+        log.prompt.error(msg)
+        QToolTip.showText(self._lineedit.mapToGlobal(QPoint(0, 0)), msg)
+
     def accept(self, value=None):
         text = value if value is not None else self._lineedit.text()
+        text = downloads.transform_path(text)
+        if text is None:
+            self._show_error("Invalid filename")
+            return False
```

Comment:
I've added it for completeness's sake, since all filename prompts should probably behave the same.

Thinking about it, putting the whole logic into the base class (`FilenamePrompt`) would probably be good, since the functions (`accept`) basically do the same - one just wraps the result in a `downloads.FileDownloadTarget`.